src/hooks/use-persist.ts

```diff
@@ -0,0 +1,30 @@
+import {
+  Dispatch, SetStateAction, useEffect, useRef,
+} from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { Persist } from '../actions/persist.actions';
+import { PersistSelector } from '../selectors/persist.selector';
+
+export default function usePersist<T>(persistKey: string | undefined, [value, setValue]: [T, Dispatch<SetStateAction<T>>]): [T, Dispatch<SetStateAction<T>>] {
```

Comment:
That's how I had things at first, but my goal was to make it composable since I needed the useDeepEquality() mechanism. That said, I ended up working around that issue so I could go with the simpler API.